### DISCUSSION TITLE
Does not work on Windows 7

### DISCUSSION THREAD

**Discussion by sergeevabc** [helix-editor/helix#9025] [Troubleshooting] [answered]
Helix 23.10, Windows 7 x64

``` console
$ hx.exe
Error: Bracketed paste not implemented in the legacy Windows API.
```

Why?

**Comment by TornaxO7** [comment]
[Any kind of support for windows 7 has been dropped](https://www.microsoft.com/en-us/windows/end-of-support#:~:text=After%2010%20years%2C%20security%20updates,the%20next%20steps%20with%20ease.)

In my opinion, Helix is not responsible for working on deprecated systems. For example, assuming there's a security issue in Windows 7 which can be exploited due to Helix, how should the Helix-Team fix it, if the actual issue resides in Windows 7 and not Helix itself?

Also as the error says, it looks like that Windows 7 uses a deprecated API in comparison to Windows 10 and Windows 11.

**Reply by sergeevabc** [reply]
Who cares what Microsoft supports or not? They do this or that for a financial reason: accelerating obsolescence, forcing patch addiction (SaaS), and EOL fear-mongering of shit-scared zoomers. In short, they extort money for zero breakthroughs (e.g. Win 10 offers nothing compared to what NTFS was after FAT32, however is full of a tracking nightmare). I have a hammer made more than 30 years ago by a manufacturer that's long gone, yet it's still good for hammering nails, without any support needed -- some things reach a state where they are developed enough, i.e. when they are complete in a necessary and sufficient way. Look around: even Powershell (7.2.17) is still built for Windows 7, not to mention popular apps such as Autohotkey, Foobar2000, Irfanview, KeePass, LibreOffice, MPC-HC, SublimeText, SumatraPDF, Telegram, etc.

How is it that Helix, out of the blue, needs something missing in Windows 7, although other text editors (including the two mentioned above) have enough?

**Reply by pascalkuthe** [reply, marked as answer]
Rust will stop support windows 7 soon (2024/02) and there is nothing we can do at that point. We will not backport a compiler/stick to an old compiler version forever because a single person like to use an insecure unspported OS.

Since it will be impossible to support windows 7 in 2 months anyway there is no point in investigating an alternative API now.

The reason we (and similarly the rust projectl care whether an OS is supported is because it takes a lot of work (our free time) and resources (CI) to support an OS. That is simply not worth our time for an ancient EOL OS that only a handful of people still use.

**Comment by kirawi** [comment]
I don't believe we make any assumptions that Windows supports bracketed paste. So it's probably crashing on Windows's side when we are querying the terminal to see if it supports it.

**Comment by gnietto** [comment]
> Rust will stop support windows 7 soon (2024/02) and there is nothing we can do at that point. We will not backport a compiler/stick to an old compiler version forever because a single person like to use an insecure unspported OS.
> 
> Since it will be impossible to support windows 7 in 2 months anyway there is no point in investigating an alternative API now.
> 
> The reason we (and similarly the rust projectl care whether an OS is supported is because it takes a lot of work (our free time) and resources (CI) to support an OS. That is simply not worth our time for an ancient EOL OS that only a handful of people still use.

My computer works with Windows 8.1 and running hx also shows "Error: Bracketed paste not implemented in the legacy Windows API." message. It's for the same reason?

**Reply by the-mikedavis** [reply]
The error comes from here when we setup the terminal: https://github.com/helix-editor/helix/blob/f27fdb2bf4da1e9b1e0a0c8e7688e738379ba33b/helix-tui/src/backend/crossterm.rs#L134-L139

To prevent the exit on startup we could handle the `Unsupported` error from crossterm (https://github.com/crossterm-rs/crossterm/blob/08762b3ef4519e7f834453bf91e3fe36f4c63fe7/src/event.rs#L390-L395) instead and log that helix won't be able to use bracketed paste

**Reply by archseer** [reply]
I think that's a reasonable tradeoff: it won't break the editor altogether and is best effort
